app/models/cd_api/base_model.rb

```diff
@@ -3,11 +3,21 @@
 require 'active_resource'
 module CdApi
   class BaseModel < ActiveResource::Base
+    cattr_accessor :static_headers
+    self.static_headers = headers
+
     self.site = Rails.configuration.x.court_data_api_config.uri
     self.user = Rails.configuration.x.court_data_api_config.user
     self.password = Rails.configuration.x.court_data_api_config.secret
     self.include_format_in_path = false
     self.format = :json
     headers['Content-Type'] = 'application/json'
+    headers['Laa-Transaction-Id'] = SecureRandom.uuid
+
+    def self.headers
+      new_headers = static_headers.clone
+      new_headers['Laa-Transaction-Id'] = SecureRandom.uuid
+      new_headers
+    end
   end
```

Comment:
This could achieve the same thing without using the clone method:

```
    ...
    self.include_format_in_path = false
    self.format = :json
    
    def self.headers
        {
          "Content-Type" => 'application/json',
          'Laa-Transaction-Id'=> SecureRandom.uuid
        }
    end
```
also means you can remove all the below:
```
    headers['Content-Type'] = 'application/json'
    headers['Laa-Transaction-Id'] = SecureRandom.uuid
    cattr_accessor :static_headers
    self.static_headers = headers
```

Reply:
I was trying to find the cleanest way of doing this and well, you have saved the day once again!